exporter/datadogexporter/factory.go

```diff
@@ -225,6 +237,14 @@ func (f *factory) createDefaultConfig() component.Config {
 			HostnameSource: HostnameSourceConfigOrSystem,
 		},
 	}
+	if isLogsAgentExporterEnabled() {
+		// logs agent sets this address by default to "https://agent-http-intake.logs.datadoghq.com"
+		cfg.Logs.TCPAddrConfig.Endpoint = ""
+		cfg.Logs.UseCompression = true
+		cfg.Logs.CompressionLevel = 6
+		cfg.Logs.BatchWait = 5
```

Comment:
Why not set these directly in the config struct above? Why do we care about the flag being enabled for this?

Reply:
These config options are only valid when the feature flag is enabled, so we don't want to set them by default otherwise.

Reply:
If we do things via unmarshaling, this should not be an issue and we can simplify the code a bit